erdjs/src/dapp/hwProvider.ts

```diff
@@ -91,14 +91,21 @@ export class HWProvider implements IDappProvider {
         const address = await this.getCurrentAddress();
         transaction.sender = new Address(address);
 
-        const sig = await this.hwApp.signTransaction(transaction.serializeForSigning(new Address(address)));
-        transaction.signature = new Signature(sig);
+        transaction.signature = await this.sign(transaction.serializeForSigning(new Address(address)));
 
         await transaction.send(this.provider);
 
         return transaction;
     }
 
+    async sign(buffer: Buffer): Promise<Signature> {
+        if (!this.hwApp) {
+            throw new Error("HWApp not initialised, call init() first");
+        }
+
+        return new Signature(await this.hwApp.signTransaction(buffer));
```

Comment:
This is not really correct - but this is something missing on our side. Like @AdoAdoAdo mentioned, the ledger app already has the message signing functionality, which is a bit different from transaction signing (e.g the prepended message). But the wrapper over that functionality is not yet implemented in `@elrondnetwork/hw-app-elrond`. I created an issue so we can link to it here: https://github.com/ElrondNetwork/hw-app-elrond/issues/4 
If you'd like, you can implement it yourself, but you'd need a ledger for testing it. If not, we can update this PR once we implement it

Reply:
I'd love to work on it but unfortunately that is a private project.
I used `signTransaction` because it actually represents a generic `signBuffer` functionality, rather than a specific transaction (it doesn't know what a transaction is anyway). Correct, if I continue to use this function, I would need to add the prefix here as well. But if you want to be more specific in regards to message signing, I need access to that repo or wait until `signMessage` is exposed.

Reply:
Yup missed that it's still a private repo. But we need to wait for the library functionality, we cannot use `signTransaction`. Even though it receives a buffer, the ledger app deserializes that buffer and checks for some mandatory fields. It also uses the deserialized values to display the transaction fields on the screen before the user confirms the signing on the device.

Reply:
Ohhh, ok. Sorry, but I didn't see this code :)

Reply:
@alexandru-g `@elrondnetwork/hw-app-elrond v0.3.0` was published, you can now update it in `package.json` and use the `signMessage` method :)